src/main/java/uk/gov/companieshouse/items/orders/api/controller/CertificateItemsController.java

```diff
@@ -22,29 +24,40 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAMESPACE);
 
     private final CreateItemRequestValidator validator;
-
-    public CertificateItemsController(CreateItemRequestValidator validator) {
+    private final CertificateItemMapper mapper;
+    private final CertificateItemService service;
+
+    /**
+     * Constructor.
+     * @param validator the validator this relies on for some 'input' validations
+     * @param mapper mapper used by this to map between {@link CertificateItemDTO} and
+     *               {@link CertificateItem} instances
+     * @param service the service used by this to manage and store certificate items
+     */
+    public CertificateItemsController(final CreateItemRequestValidator validator,
+                                      final CertificateItemMapper mapper,
+                                      final CertificateItemService service) {
         this.validator = validator;
+        this.mapper = mapper;
+        this.service = service;
     }
 
     @PostMapping("${uk.gov.companieshouse.items.orders.api.path}")
-    public ResponseEntity<Object> createCertificateItem(final @Valid @RequestBody CertificateItemDTO certificateItemDTO) {
-
+    public ResponseEntity<Object> createCertificateItem(final @Valid @RequestBody CertificateItemDTO certificateItemDTO)
+    {
         LOGGER.info("ENTERING createCertificateItem(" + certificateItemDTO + ")");
 
         final List<String> errors = validator.getValidationErrors(certificateItemDTO);
         if (!errors.isEmpty()) {
             return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new ApiError(HttpStatus.BAD_REQUEST, errors));
         }
 
-        CERTIFICATE.populateReadOnlyFields(certificateItemDTO);
-
-        // TODO PCI-324 ID will be generated as per
-        //  https://companieshouse.atlassian.net/wiki/spaces/DEV/pages/1258094916/Certificates+API+High+Level+Development+Design#CertificatesAPIHighLevelDevelopmentDesign-Traceability
-        certificateItemDTO.setId("CHS1");
+        CertificateItem item = mapper.certificateItemDTOtoCertificateItem(certificateItemDTO);
+        item = service.createCertificateItem(item);
+        final CertificateItemDTO createdCertificateItemDTO = mapper.certificateItemToCertificateItemDTO(item);
 
-        LOGGER.info("EXITING createCertificateItem() with " + certificateItemDTO);
-        return ResponseEntity.status(HttpStatus.CREATED).body(certificateItemDTO);
+        LOGGER.info("EXITING createCertificateItem() with " + createdCertificateItemDTO);
```

Comment:
Other applications have used a LoggerInterceptor to log the requests and responses (with the status code) for all requests without log messages throughout the code.

Reply:
👍 Yes, thanks Joanne. I will be looking into that as part of PCI-478 Use structured logging for consistent CH ELK stack compatible logging.